SalesforceCommerceCloud/pwa-kit
# Grab correlation ID and send as a response header

# Description

As part of log forwarding, we are now generating a v4 UUID Correlation ID to attach to our MRT logs so that our logs can be searched for outside of MRT. In MRT, this is now added as a new header, `x-correlation-id`, to the event object, which is the json representation of the HTTP request. In pwa-kit, we then need to pull this new header, modify the response, and add this header to the response so that the correlation ID value is accessible and visible to the user.

GUS: [W-15706159](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001qpcAeYAI/view)
Linked PR to generate Correlation ID and add to Log Center logs: https://git.soma.salesforce.com/cc-mobify/portal_app/pull/5770
Linked PR to add Correlation ID to Splunk logs and log tailing: https://git.soma.salesforce.com/cc-mobify/portal_app/pull/5806

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- In [_setRequestId()](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1787/files#diff-e2b609f0f685de7f6edc3064b095881faa48d282748781225c8732e363a7397eR255), instead of the `x-api-gateway-event` header, the `x-correlation-id` header is now the value used to set the new requestId
- In [processLambdaResponse()](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1787/files#diff-1b02b69d1b5dab55ffd25b21f65a6b78bd4999ba53e9c86a7211502df90769eaR10), `x-correlation-id` is set as a response header if it exists

# How to Test-Drive This PR

- Navigate to https://yuna-production.mobify-storefront-staging.com , which has the mrt-template-reference-app deployed to it with the changes from this branch
- Send a request
- Verify that the `x-correlation-id` is set as a response header
<img width="1630" alt="Screenshot 2024-06-03 at 4 08 21 PM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/84923642/d453e448-475b-4b1f-89d0-c93b68f94bee">

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
